.github/workflows/pin-build-tools-image.yml

```diff
@@ -33,10 +33,6 @@ concurrency:
 # No permission for GITHUB_TOKEN by default; the **minimal required** set of permissions should be granted in each job.
 permissions: {}
 
-env:
-  FROM_TAG: ${{ inputs.from-tag }}
-  TO_TAG: pinned
-
```

Comment:
Inlining these env vars because `jobs.<job_id>.with.<with_id>` doesn't have the `env` context available, see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#context-availability

Reply:
I'd still keep and use where we can, for example in steps that run shell scripts. It makes it easier to copy and run script locally without changing too much, but just setting FROM_TAG/TO_TAG vars.
Also using `inputs` right in the shell potentially leads could lead to injections: https://codeql.github.com/codeql-query-help/javascript/js-actions-command-injection/#example

Reply:
Agreed. I'll add it back, but will add a comment explaining that the image map needs updating as well if the env is updated, so that it's harder to accidentally miss that. That's the reason I removed it instead of going for the mixed approach you're suggesting.